oshmem/mca/scoll/basic/scoll_basic_alltoall.c

```diff
@@ -61,6 +61,11 @@ int mca_scoll_basic_alltoall(struct oshmem_group_t *group,
         return OSHMEM_ERR_BAD_PARAM;
     }
 
+    /* Do nothing on zero-length request */
+    if (OPAL_UNLIKELY(!nelems)) {
```

Comment:
let's double check what is the synchronization expectation from shmem collectives which are not barrier